include/CL/opencl.hpp

```diff
@@ -5838,6 +5838,15 @@ class NDRange
         sizes_[2] = size2;
     }
 
+    //! \brief Constructs one-dimensional range.
+    NDRange(std::array<size_type, 1> a) : NDRange(a[0]){}
+
+    //! \brief Constructs two-dimensional range.
+    NDRange(std::array<size_type, 2> a) : NDRange(a[0], a[1]){}
+
+    //! \brief Constructs three-dimensional range.
+    NDRange(std::array<size_type, 3> a) : NDRange(a[0], a[1], a[2]){}
```

Comment:
I think we should use the un-namespaced `array` class here rather than `std::array`, to work with `CL_HPP_NO_STD_ARRAY`.

```suggestion
    //! \brief Constructs one-dimensional range.
    NDRange(array<size_type, 1> a) : NDRange(a[0]){}

    //! \brief Constructs two-dimensional range.
    NDRange(array<size_type, 2> a) : NDRange(a[0], a[1]){}

    //! \brief Constructs three-dimensional range.
    NDRange(array<size_type, 3> a) : NDRange(a[0], a[1], a[2]){}
```

Reply:
Good point. Thank you!